foundations/html_css/html-foundations/commit-messages.md

```diff
@@ -0,0 +1,50 @@
+### Introduction
+
+This lesson will explain the importance of good commit messages, how to write them, when to commit and why having a history of good commits is so important!
+
+### Lesson Overview
+This section contains a general overview of topics that you will learn in this lesson.
+- How to write a meaningful commit message
+- Why meaningful commit messages are important
+- When to commit
+
+### Are commit messages so important they deserve their own lesson?
+
+Yes! Let me give you a quick list of reasons why:
+
+- When you are applying for jobs, employers will look through your projects on GitHub and they will look through your commit history. Having good commits as a novice developer will help you stand out. 
+- Having a good commit message history will allow you (or other developers working on your code) to quickly see what changes were made and why. This is useful if a bug is found in the code that needs to be fixed!
+- Having a good commit message history will also be helpful if you come back to a project you were working on after stepping away from it for a while. You likely won't remember your thought process and changes made when initially writing out the code. 
+
+### Assignment 
+
+<div class="lesson-content__panel" markdown="1">
+
+1.  This article, [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/), covers all the main bases on how to write good commit messages. The whole article is great and informative but the meat of the article is “The seven rules of a great commit message.”
+
+</div>
+
+
+
```

Comment:
I believe that the assignment should come near the end of the lesson right before the knowledge check.

Reply:
I moved it down but put left "Tips and things to remember" after it since that is all stuff from that article

Reply:
@thatblindgeye do you have any thoughts on this? from a style guide perspective? If we have additional first-party content that references the assignments is this the pattern we should follow?

Reply:
As long as the post-Assignment content isn't something *necessary* for lesson completion (which in this case it isn't; it's just a "remember..." sort of section), I'd say this should be fine. 

The thing with the style guide is it should be a "lessons should follow this guide *as closely as possible*" sort of thing. Some lessons needing to deviate a bit like in this instance I think is fine, as long as it's nothing completely wild (like having the assignment come before the LO section or something).

If anything we could also add a post-Assignment section to the lesson layout (similar to how the project layout has), but since this may be more rare a thing my initial thought is we don't really *need* to (but that's just me).